modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -1245,6 +1246,52 @@ class DeConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
 
         pad.width = pad_l;
         pad.height = pad_t;
+
+        const int inpCn = inputs[0].size[1];
+        const int outCn = numOutput;
+        weightsMultipliers.assign(outCn, 1.0);
+        if (weightsMat.empty())
+        {
+            weightsMat = (blobs[0].reshape(1, inpCn)).clone();
+            biasesMat = hasBias() ? blobs[1].reshape(1, outCn) : Mat::zeros(outCn, 1, CV_32F);
+        }
+    }
+
+    virtual bool tryFuse(Ptr<Layer>& top) CV_OVERRIDE
+    {
+        Mat w, b;
+        top->getScaleShift(w, b);
+        if (!w.empty() || !b.empty())
+        {
+            fuseWeights(w, b);
+            return true;
+        }
+        return false;
+    }
+
+    void fuseWeights(const Mat& w_, const Mat& b_)
+    {
+        const int outCn = numOutput;
+        Mat w = w_.total() == 1 ? Mat(1, outCn, CV_32F, Scalar(w_.at<float>(0))) : w_;
+        Mat b = b_.total() == 1 ? Mat(1, outCn, CV_32F, Scalar(b_.at<float>(0))) : b_;
+        CV_Assert_N(!weightsMat.empty(), w.empty() || outCn == w.total(), b.empty() || outCn == b.total());
+        if (!w.empty())
+        {
+            if (weightsMat.data == blobs[0].data) { weightsMat = weightsMat.clone(); }
+            Mat originWeights = weightsMat.clone();
+            for (int i = 0; i < outCn; ++i)
+            {
+                double wi = w.at<float>(i);
+                weightsMultipliers[i] *= wi;
+                cv::multiply(originWeights.row(i), weightsMultipliers[i], weightsMat.row(i));
```

Comment:
That looks confusing. `weightsMat` is initialized with as `weightsMat = (blobs[0].reshape(1, inpCn)).clone();` (with number of rows equals number of input channels) but in this loop we iterate over rows by number output channels.